crates/net/network/src/session/mod.rs

```diff
@@ -264,6 +265,18 @@ impl SessionManager {
         }
     }
 
+    /// Sends a disconnect message with  TooManyPeers reason.
+    ///
+    /// Called when We currently don't have additional capacity to establish a session from an
+    /// incoming connection.
+    pub(crate) fn disconnect_incoming_connection(&self, stream: TcpStream) {
```

Comment:
I'd like this to accept the DisconnectReason so we can do the same with `UselessPeer` for example